app/modules/repo_manager/repo_manager.py

```
@@ -0,0 +1,511 @@
+        self,
+        repo_name: str,
+        local_path: str,
+        branch: Optional[str] = None,
+        commit_id: Optional[str] = None,
+        user_id: Optional[str] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+    ) -> str:
+        """Register a repository that has been downloaded/parsed."""
+        # Validate local path exists
+        if not os.path.exists(local_path):
+            raise ValueError(f"Local path does not exist: {local_path}")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)